src/java.base/share/classes/java/util/concurrent/DelayScheduler.java

```diff
@@ -393,6 +398,7 @@ else if (leastChild == null || cd < leastValue) {
                 k = alsoReplace;
             }
         }
+        assert checkHeap(h, n);
```

Comment:
It's great to have this run during debug builds :+1:

Reply:
The tests runs are done with -esa so the assert is always enabled in test runs. So all good.